Code/Editor/Plugins/ProjectSettingsTool/ProjectSettingsContainer.cpp

```diff
@@ -87,36 +89,45 @@ namespace ProjectSettingsTool
     }
 
 
-    ProjectSettingsContainer::ProjectSettingsContainer(const AZStd::string& projectJsonFileName, PlistInitVector& plistPaths)
+    ProjectSettingsContainer::ProjectSettingsContainer(const AZStd::string& projectJsonFileName, const PlatformResources& platformResources)
         : m_projectJson(JsonSettings{ projectJsonFileName, "", AZStd::make_unique<rapidjson::Document>() })
     {
-        LoadProjectJsonData();
+        LoadJson(m_projectJson);
 
-        for (PlatformAndPath& plistInfo : plistPaths)
+        for (const auto& [platformId, path] : platformResources)
         {
-            m_pListsMap.insert(AZStd::pair<PlatformId, PlistSettings>(
-                plistInfo.first,
-                PlistSettings{ plistInfo.second, "", AZStd::make_unique<XmlDocument>() }));
+            if (AZ::IO::PathView(path).Extension() == ".json")
+            {
+                m_platformSettingsMap.insert(AZStd::pair<PlatformId, PlatformSettings>(
+                    platformId,
+                    JsonSettings{ path, "", AZStd::make_unique<rapidjson::Document>() }));
+
+                LoadJson(AZStd::get<JsonSettings>(m_platformSettingsMap[platformId]));
+            }
+            else
+            {
+                m_platformSettingsMap.insert(AZStd::pair<PlatformId, PlatformSettings>(
+                    platformId,
+                    PlistSettings{ path, "", AZStd::make_unique<XmlDocument>() }));
 
-            // We will have to find it then because unique_ptrs are not copy constructible
-            // And a move constructor would copy large strings
-            LoadPlist(m_pListsMap.find(plistInfo.first)->second);
+                LoadPlist(AZStd::get<PlistSettings>(m_platformSettingsMap[platformId]));
+            }
         }
     }
 
     ProjectSettingsContainer::~ProjectSettingsContainer()
     {
     }
 
-    ProjectSettingsContainer::PlistSettings* ProjectSettingsContainer::GetPlistSettingsForPlatform(const Platform& plat)
+    ProjectSettingsContainer::PlatformSettings* ProjectSettingsContainer::GetPlatformData(const Platform& plat)
```

Comment:
Could this function be made `const`?

Reply:
No because it's returning a non-const pointer to its data.